### PR TITLE
Indentation

### PR DESCRIPTION
**Pull request by dphov**
This PR holding changes for generating "readable" code with indentation

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -16,18 +16,29 @@ extern crate rustc_middle;
...
 
 use pretty::RcDoc;
+
+mod render;
 use rustc_errors::registry;
 use rustc_session::config::{self, CheckCfg};
 use rustc_span::source_map;
 
+#[derive(Debug)]
+
```
Comment (clarus) [inline src/main.rs:29]:
```suggestion
```

File: src/render.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+use pretty::RcDoc;
+
+pub(crate) fn paren(with_paren: bool, doc: RcDoc<()>) -> RcDoc<()> {
+    if with_paren {
+        RcDoc::text("(").append(doc).append(RcDoc::text(")"))
```
Comment (clarus) [inline src/render.rs:5]: Maybe we can use https://docs.rs/pretty/latest/pretty/struct.RcDoc.html#method.concat to reduce the number of `append` and with `use pretty::RcDoc` avoid the need for the prefix `RcDoc`.

File: src/render.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    } else {
+        doc
+    }
+}
+
+pub(crate) fn bracket(doc: RcDoc<()>) -> RcDoc<()> {
```
Comment (clarus) [inline src/render.rs:11]: OK I did not know the `pub(crate)`.

Comment (dphov) [reply src/render.rs:11]: `pub(crate)` will set _crate only_ visibility for a function.

File: src/render.rs (hunk 2 above)
Comment (dphov) [reply src/render.rs]: Resolved in https://github.com/formal-land/coq-of-rust/pull/6/commits/d2dc19a81f56ea548cf2b6ff72af83ba6ecd15c0 commit

**Pull request merged**